### ISSUE TITLE
Path winding number is -1 when point is horizontal to path point

### ISSUE THREAD

**Issue by jpopesculian** [nical/lyon#636]
I encountered a small issue with doing the hit_test when a point is horizontal to a point in the path in which one is testing. As an example the following returns a -1 winding number and therefore returns that the point is inside the path although it is clearly not

```rust
use lyon::algorithms::hit_test::path_winding_number_at_position;
use lyon::math::point;
use lyon::path::polygon::Polygon;

fn main() {
    let square = Polygon {
        points: &[
            point(-10., 10.),
            point(10., 10.),
            point(10., 5.),
            point(10., -10.),
            point(-10., -10.),
        ],
        closed: true,
    };

    assert_eq!(
        -1,
        path_winding_number_at_position(&point(15., 5.), square.path_events(), 0.)
    )
}
```

**Comment by nical** [comment]
Thanks for filing this, I have a fix coming soon.

**Comment by nical** [comment]
The fix is published in `lyon_algorithms 0.17.3`.

**Comment by jpopesculian** [comment]
Oh wow! Crazy turn around! Thanks
